### PR TITLE
Feat: UI Add Support for @ and # in text editor

### LINKED ISSUE #2997: UI: Add Support  for @ and # in text editor
Refer to mocks in #2645

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Fixes #2997 
### Describe your changes :

See issue #2997 

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/156186123-fe0fe014-c06a-42b9-a744-0dbde7163217.mov

cc : @harshach @sureshms @vivekratnavel 
#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t, @darth-coder00

### CODE REVIEW

**Review by darth-coder00 — APPROVED** [review]
LGTM!

**Review by vivekratnavel — APPROVED** [review]

**Comment by vivekratnavel** [comment]
The following Selenium test fails
```
Error:  Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 70.313 s <<< FAILURE! - in org.openmetadata.catalog.selenium.pagesWithoutData.myData.MyDataPageTest
Error:  org.openmetadata.catalog.selenium.pagesWithoutData.myData.MyDataPageTest.checkOverview  Time elapsed: 8.27 s  <<< FAILURE!
```
I am ok to merge and take care of it in a separate issue.

File: openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts [hunk 1]
```diff
@@ -95,3 +108,125 @@ export const getEntityFieldThreadCounts = (
...
+) {
+  const matches = await suggestions(searchTerm, mentionChar);
+  renderList(matches, searchTerm);
+}
+
+const getMentionList = (message: string) => {
+  return message.match(mentionRegEx);
+};
+
+const getHahsTagList = (message: string) => {
```
Comment (mithmatt) [inline openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts:183]: HashTag instead of HahsTag

**Review by mithmatt — COMMENTED** [review]
For the entity mention we use #. 
Can # be used for tags too?

**Comment by Sachin-chaurasiya** [comment]
> For the entity mention we use #. Can # be used for tags too?

Hey @mithmatt , No, we can only mention entity.

**Comment by mithmatt** [comment]
> > For the entity mention we use #. Can # be used for tags too?
> 
> Hey @mithmatt , No, we can only mention entity.

Something we can add in the future..

File: openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/FeedUtils.ts]: Thanks @mithmatt , for catching this typo.

**Review by mithmatt — APPROVED** [review]
Looks good besides the test failure

**Comment by Sachin-chaurasiya** [comment]
> > > For the entity mention we use #. Can # be used for tags too?
> > 
> > 
> > Hey @mithmatt , No, we can only mention entity.
> 
> Something we can add in the future..

Yeah, we need to come up with something so user would not get confusion between entities and tags.

> > > For the entity mention we use #. Can # be used for tags too?
> > 
> > 
> > Hey @mithmatt , No, we can only mention entity.
> 
> Something we can add in the future..

Yeah, we need to come up with something so user would not get confusion between entities and tags.

**Pull request merged**